service/src/main/java/org/cbioportal/service/impl/GenesetDataServiceImpl.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2016 The Hyve B.V.
+ * This code is licensed under the GNU Affero General Public License (AGPL),
+ * version 3, or (at your option) any later version.
+ */
+
+/*
+ * This file is part of cBioPortal.
+ *
+ * cBioPortal is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package org.cbioportal.service.impl;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.cbioportal.model.GenesetGeneticAlteration;
+import org.cbioportal.model.GenesetGeneticData;
+import org.cbioportal.model.GeneticProfile;
+import org.cbioportal.model.Sample;
+import org.cbioportal.persistence.GeneticDataRepository;
+import org.cbioportal.persistence.SampleListRepository;
+import org.cbioportal.service.GenesetDataService;
+import org.cbioportal.service.GeneticProfileService;
+import org.cbioportal.service.SampleService;
+import org.cbioportal.service.exception.GeneticProfileNotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class GenesetDataServiceImpl implements GenesetDataService {
+
+    @Autowired
+    private GeneticDataRepository geneticDataRepository;
+    @Autowired
+    private SampleService sampleService;
+    @Autowired
+    private GeneticProfileService geneticProfileService;
+    @Autowired
+    private SampleListRepository sampleListRepository;
```

Comment:
Can we use SampleListService here instead of using SampleListRepository directly?